### ISSUE TITLE
Mean image

### ISSUE THREAD

**Issue by FengLoveBella** [hellochick/PSPNet-tensorflow#10]
May be the mean image in your code is fault. The mean image is Pascal.

**Comment by hellochick** [comment]
Yes, @zhoufengbuaa . I know the mean of image I used is for Pascal. But I think image mean is used for shifting the RGB value to -128~128. So the value of mean to subtract may have a  small influence. Do you agree what I say ? Thanks

**Comment by FengLoveBella** [comment]
Agree

**Comment by FengLoveBella** [comment]
@hellochick I use your code and your model, but I got only 0.576 accuracy. Did you really got 77.23% accuracy on validation dataset of cityscape dataset ??

**Comment by FengLoveBella** [comment]
![screenshot from 2017-12-12 09-39-09](https://user-images.githubusercontent.com/11606333/33863139-585e6f0e-df21-11e7-9a71-15635b60546c.png)

**Comment by FengLoveBella** [comment]
And it is much poor than deeplab v2, I used deeplab v2, I got 70+ accuracy. @hellochick

**Comment by John1231983** [comment]
@zhoufengbuaa : Very nice report. Could you tell me how did you achieve the report? Thanks

**Comment by FengLoveBella** [comment]
please refer to the python code provided by cituscape dataset. @John1231983

**Comment by FengLoveBella** [comment]
what IOU did you get, I used the model I only get IOU about 0.5~0.6 per 10 steps. @hellochick @John1231983

**Comment by John1231983** [comment]
@zhoufengbuaa : Mee too. When I used hellochick code I only got around 60% IoU. I guess something wrong when I run the code. But my question is how could you generate the nice report as the figure? Do you use some script or python code for making the report?

**Comment by hellochick** [comment]
@zhoufengbuaa @John1231983 , Did you run the `evaluate.py` with  `trainIdLabelImg` ? You should first transform the images using this script: https://github.com/mcordts/cityscapesScripts/blob/master/cityscapesscripts/preparation/createTrainIdLabelImgs.py
Please tell me your evaluation procedure, thanks. Btw, I really got that result ( Without filpping ):
![default](https://user-images.githubusercontent.com/18046598/33868305-27a69cee-df3d-11e7-9e1d-13bd57c96df2.JPG)

**Comment by zhoufengbuaa** [comment]
I am sorry, it's my fault, I should not doubt your code, and I got your accuracy, and it's about 0.77. The reason is caused by mean image.  Thank you very much. 
![Uploading Screenshot from 2017-12-12 16-38-11.png…]()

**Comment by FengLoveBella** [comment]
I am sorry, it's my fault, I should not doubt your code, and I got your accuracy, and it's about 0.77. The reason is caused by mean image. Thank you very much.

![screenshot from 2017-12-12 16-38-11](https://user-images.githubusercontent.com/11606333/33875002-c68adc44-df5b-11e7-8ea8-38019097c407.png)

**Comment by FengLoveBella** [comment]
please refer to the evaluation python code provided by cityscape dataset, you just need put you predition result to this code, and it will give you this output. @John1231983

**Comment by smmdream** [comment]
![2017-12-14 16-00-49](https://user-images.githubusercontent.com/10106222/33981521-32218efc-e0e8-11e7-9563-5bfd3f838227.png)
@hellochick @zhoufengbuaa  This is my evaluation result, the mIoU is gradually increasing, only 0.502 at the beginning , is it right or normal?

**Comment by hellochick** [comment]
@smmdream , yes it's normal. Since not all objects appear in the beginning (One image may not contain 19 classes together), so the mIoU will gradually increase.

**Comment by smmdream** [comment]
@hellochick , I find a problem that the same model, the first evaluation result is different from the second evaluation result and the third and so on, and their value are quite different. Usually the first evaluation result is better.
![2017-12-14 20-51-42](https://user-images.githubusercontent.com/10106222/33993311-efd54e02-e110-11e7-9453-fda5bcfb0372.png)
![2017-12-14 20-52-46](https://user-images.githubusercontent.com/10106222/33993318-f3f10d1e-e110-11e7-86b6-9bf34201d81d.png)
![2017-12-14 20-53-39](https://user-images.githubusercontent.com/10106222/33993323-f7efb078-e110-11e7-9d27-b041cf2fd7f8.png)

**Comment by hellochick** [comment]
Hey, since you use different part of queue, the result will be different. But if you run the whole evaluation data, the result will be the same.

**Comment by smmdream** [comment]
@hellochick , oh I see, I do the test, you are right.

**Comment by smmdream** [comment]
@zhoufengbuaa I want to get the mIoU of each category, such as yours, can you tell me how to get that?

**Comment by rydeldcosta** [comment]
@zhoufengbuaa Hey, you said you made a mistake with the mean image which caused you to get low mIoU. Can you please elaborate ? I am facing the same issue

**Comment by FengLoveBella** [comment]
@rydeldcosta mean image, just use the mean image provided in this code.

**Comment by xlmou** [comment]
@zhoufengbuaa Hi friends, where can I get an initial model pretrained on ImageNet to train PSPnet ? Can you give me a download link ?

**Comment by ga84** [comment]
@zhoufengbuaa @hellochick
I know it's been a long time since you posted here, but could you please give me some advice on how to get images with (class - or training) label IDs? In order to evaluate with cityscapes scripts you must have had these, right? So can you please tell me in which way you got them? I tried with this code by getting IDs as: IDs = sess.run(raw_output_up) however this gives an image with numbers which don't represent neither train IDs, nor class labels (numbers up to 255, but don't fit to these two IDs)....

**Comment by LbinB** [comment]
Did you get a solution about the problem? @ga84

**Comment by Jason-xin** [comment]
> @zhoufengbuaa I want to get the mIoU of each category, such as yours, can you tell me how to get that?

So do I? Have you solved yet?
